### ISSUE TITLE
SmartComponentsScriptTagHelperComponent duplicates JS when Blazor initialized from Razor Page

### ISSUE THREAD

**Issue by hakenr** [dotnet-smartcomponents/smartcomponents#9]
We are still using "ASP.NET Core Hosted Blazor WebAssembly with server prerendering" (the pre-BWA setup):
* `_Layout.cshtml` + `_Host.cshtml` host the `<component type="typeof(App)" ... />`
* The `SmartComponentsScriptTagHelperComponent` registered in `AddSmartComponents()` adds `<script />` reference to the JS which duplicates the library module initializer added by Blazor:
![image](https://github.com/dotnet-smartcomponents/smartcomponents/assets/12828888/5513f421-8136-4d86-a7b9-fee22ccc01da)

Workaround:
```csharp
services.Remove(services.Single(s => s.ImplementationType?.Name == "SmartComponentsScriptTagHelperComponent"));
```
(after `services.AddSmartComponents();`)

cc @SteveSandersonMS

**Comment by SteveSandersonMS** [comment]
Thanks for reporting it. I guess there are two main possible solutions:

1. Stop injecting the script automatically with cshtml and go back to the `<smart-components-script>` tag to inject the JS
2. Or, update the JS to make it tolerate being loaded multiple times, and just no-op on the 2nd and later imports

Solution 2 is probably best even though it means a duplicated HTTP request. We could even go further and allow configuring a flag to disable the auto-injection for people who are using Blazor+cshtml and want to avoid the extra HTTP request.

**Comment by SteveSandersonMS** [comment]
This error should now be fixed in version [0.1.0-preview10141](https://www.nuget.org/packages/SmartComponents.AspNetCore/0.1.0-preview10141) which is now on NuGet.

Please let me know if you still run into it!

**Comment by hakenr** [comment]
Thanks for the quick response.

Confirmed. The error in the browser console is gone; the script gets loaded twice (once with a version query from the Razor page, and once without the version from the Blazor startup).

I understand the tradeoff here. Given that "Blazor hosted from cshtml" will become increasingly rare (we will be moving to Blazor WebAssembly soon), the solution seems acceptable for now.
